ios/ReactNativeSigner.swift

```diff
@@ -14,11 +14,18 @@ class ReactNativeSigner: NSObject, XMTP.SigningKey {
 
 	var module: XMTPModule
 	var address: String
+	var walletType: WalletType
+	var chainId: Int64?
+	var blockNumber: Int64?
 	var continuations: [String: CheckedContinuation<XMTP.Signature, Swift.Error>] = [:]
+	var scwContinuations: [String: CheckedContinuation<Data, Swift.Error>] = [:]
 
-	init(module: XMTPModule, address: String) {
+	init(module: XMTPModule, address: String, walletType: WalletType = WalletType.EOA, chainId: Int64? = nil, blockNumber: Int64? = nil) {
 		self.module = module
 		self.address = address
+		self.walletType = walletType
+		self.chainId = chainId
+		self.blockNumber = blockNumber
```

Comment:
I don't think we need the block number to be stored on the signer. That should get resolved at the time of signing, not on instantiation, since the block number keeps incrementing.

Reply:
We store it on the `SigningKey` so it only gets passed in from the `SigningKey` which means we only have it when the signing key is present which is only when something is being signed. Unfortunately with the bridge this is how it has to be passed. But this and the below shouldn't be exposed to users so it should be an internal matter.

Reply:
OK. I can approve the PR. Still feels wrong to get that block number even one second before we actually sign the message, since it's going to be older than the block the signature was actually generated on.

Reply:
Hmmm okay maybe we need to rethink how I have this setup inside the other sdks as well. 🤔 

As they all work the same with the block number as part of the signingKey https://github.com/xmtp/xmtp-android/blob/main/library/src/main/java/org/xmtp/android/library/SigningKey.kt#L31-L34 But maybe it should be passed as a param to create instead?

Reply:
Ideally it would get pulled from the chain at the same time you are signing the message.

Reply:
Another way of thinking about it would be to have a `getBlockNumber` method that is only implemented on SCWs, which we could call from our SDKs when we are gathering a smart contract signature.